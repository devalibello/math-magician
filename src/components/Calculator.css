.calculator {
  /* border: solid 2px red; */
  margin: 20px 80px;
  height: auto;
}

.button {
  width: 25%;
  padding: 20px;
  border: 1px solid rgb(223, 219, 219);
}

.input {
  padding: 20px;
  width: 100%;
  text-align: right;
  font-size: 1.5rem;
  background-color: rgb(137, 148, 151);
  color: white;
  border: 1px solid rgb(223, 219, 219);
}

.dot {
  margin-left: auto;
}

.row-five {
  display: flex;
}

.button-zero {
  flex: 1;
}

.operator {
  background-color: orange;
}

.math-magician {
  text-align: center;
  margin-bottom: 10px;
  font-family: 'Kaushan Script', cursive, sans-serif;
  font-size: 1.5rem;
  color: rgb(88, 95, 97);
}
