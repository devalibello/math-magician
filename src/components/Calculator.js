import React, { useState, useEffect } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';
import style from '../styles/Calculator.module.css';

const Calculator = () => {
  const [calculationValues, setCalculatorValues] = useState({
    total: null,
    next: null,
    operation: null,
  });

  useEffect(() => {
    const handleButtonClick = (event) => {
      const buttonValue = event.target.textContent;
      setCalculatorValues(calculate(calculationValues, buttonValue));
    };

    const buttons = document.querySelectorAll('.button');
    buttons.forEach((button) => {
      button.addEventListener('click', handleButtonClick);
    });

    return () => {
      buttons.forEach((button) => {
        button.removeEventListener('click', handleButtonClick);
      });
    };
  });

  return (
    <div className="calculatorContainer">
      <p className={style.mathText}>Lets do some Math!!!</p>
      <div className="calculator">
        <input type="text" className="input" readOnly value={calculationValues.next || calculationValues.total || '0'} />
        <div className="buttons">
          <div className="row-one">
            <button className="button" type="button">AC</button>
            <button className="button" type="button">+/-</button>
            <button className="button" type="button">%</button>
            <button className="button operator" type="button">&#247;</button>
          </div>

          <div className="row">
            <button className="button" type="button">7</button>
            <button className="button" type="button">8</button>
            <button className="button" type="button">9</button>
            <button className="button operator" type="button">x</button>
          </div>

          <div className="row">
            <button className="button" type="button">4</button>
            <button className="button" type="button">5</button>
            <button className="button" type="button">6</button>
            <button className="button operator" type="button">-</button>
          </div>

          <div className="row">
            <button className="button" type="button">1</button>
            <button className="button" type="button">2</button>
            <button className="button" type="button">3</button>
            <button className="button operator" type="button">+</button>
          </div>

          <div className="row row-five">
            <button className="button button-zero" type="button">0</button>
            <button className="button  dot" type="button">.</button>
            <button className="button operator" type="button">=</button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
