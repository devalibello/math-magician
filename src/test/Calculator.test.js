import React from 'react';
import { render, screen } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator', () => {
  test('should render the Calculator component', () => {
    render(<Calculator />);
    expect(screen.getByTestId('calculator-input')).toMatchSnapshot();
  });
});
