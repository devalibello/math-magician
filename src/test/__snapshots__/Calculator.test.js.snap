// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should render the Calculator component 1`] = `
<input
  class="input"
  data-testid="calculator-input"
  readonly=""
  type="text"
  value="0"
/>
`;
